hr_expense_substate/views/hr_expense_views.xml

```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="view_hr_expense_sheet_tree" model="ir.ui.view">
+        <field name="model">hr.expense.sheet</field>
+        <field name="inherit_id" ref="hr_expense.view_hr_expense_sheet_tree" />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='state']" position="after">
+                <field name="substate_id" widget="badge" optional="hide" />
+            </xpath>
+        </field>
+    </record>
+    <record id="view_hr_expense_sheet_form" model="ir.ui.view">
+        <field name="model">hr.expense.sheet</field>
+        <field name="inherit_id" ref="hr_expense.view_hr_expense_sheet_form" />
+        <field name="arch" type="xml">
+            <xpath expr="//header/field[@name='state']" position="before">
+                <field
+                    name="substate_id"
```

Comment:
In other modules the substates have moved to the line below on the form view.
Example from account_move_substate I'm working on.
![image](https://github.com/OCA/hr-expense/assets/11499387/f8b4f4ca-0fae-45ad-bfa6-702f77b757a4)

https://github.com/OCA/account-invoicing/pull/1568/files#diff-1947d80641a90451e063090957b86e45ae7bbe8d67d9bea602469e8eab9e6106R8-R16